nodebuilder/rpc/config.go

```diff
@@ -20,10 +21,13 @@ func DefaultConfig() Config {
 }
 
 func (cfg *Config) Validate() error {
-	if ip := net.ParseIP(cfg.Address); ip == nil {
-		return fmt.Errorf("service/rpc: invalid listen address format: %s", cfg.Address)
+	sanitizedAddress, err := utils.ValidateAddr(cfg.Address)
+	if err != nil {
+		return fmt.Errorf("service/rpc: invalid address: %w", err)
 	}
-	_, err := strconv.Atoi(cfg.Port)
+	cfg.Address = sanitizedAddress
+
+	_, err = strconv.Atoi(cfg.Port)
```

Comment:
feels like it should be, so we don't have to remember to do the extra validation step in multiple places.